cardano-cli/src/Cardano/CLI/Environment.hs

```diff
@@ -40,13 +41,13 @@ getEnvCli = do
       , envCliAnyCardanoEra = mCardanoEra
       }
 
-envCliAnyShelleyBasedEra :: EnvCli -> Maybe (EraInEon ShelleyBasedEra)
-envCliAnyShelleyBasedEra envCli = do
+envCliSomeEra :: EnvCli -> Maybe (Exp.Some Exp.Era)
+envCliSomeEra envCli = do
   AnyCardanoEra era <- envCliAnyCardanoEra envCli
-  forEraInEonMaybe era EraInEon
+  forEraInEonMaybe era Exp.Some
 
-envCliAnyShelleyToBabbageEra :: EnvCli -> Maybe (EraInEon ShelleyToBabbageEra)
-envCliAnyShelleyToBabbageEra envCli = do
+envCliAnyEon :: Typeable eon => Eon eon => EnvCli -> Maybe (EraInEon eon)
```

Comment:
This is called in parsers like `pShelleyBasedShelley` which are not used. We can delete them.

Reply:
I was referring to parsers like `pShelleyBasedShelley` which I see have been removed already.